subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/DefaultDependencyFactory.java

```diff
@@ -111,4 +118,28 @@ private void configureModule(ClientModule clientModule, Closure configureClosure
         moduleFactoryDelegate.prepareDelegation(configureClosure);
         configureClosure.call();
     }
+
+    // region DependencyFactory methods
+
+    @Override
+    public ExternalModuleDependency createFromCharSequence(CharSequence dependencyNotation) {
+        return dependencyNotationParser.getStringNotationParser().parseNotation(dependencyNotation.toString());
+    }
+
+    @Override
+    public ExternalModuleDependency createFromMap(Map<String, ?> map) {
+        return dependencyNotationParser.getMapNotationParser().parseNotation(map);
+    }
+
+    @Override
+    public FileCollectionDependency createFromFileCollection(FileCollection fileCollection) {
+        return dependencyNotationParser.getFileCollectionNotationParser().parseNotation(fileCollection);
+    }
+
+    @Override
+    public ProjectDependency createFromProject(Project project) {
+        return dependencyNotationParser.getProjectNotationParser().parseNotation(project);
+    }
```

Comment:
Do we actually need these parsers pulled out separately? If you were to pass an invalid String to the full parser, would you get an error mentioning all of the other notations?

Reply:
I pulled them out separately for type-safety. The error is fine otherwise, e.g.:
```
Supplied String module notation 'thewrong.123898::::::format' is invalid. Example notations: 'org.gradle:gradle-core:2.2', 'org.mockito:mockito-core:1.9.5:javadoc'.
```

Should I revert this? I personally prefer to have it this way instead of casting.